src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -527,7 +527,7 @@ private BoundForEachStatement BindForEachPartsWorker(BindingDiagnosticBag diagno
 
             if (builder.InlineArraySpanType == WellKnownType.Unknown && getEnumeratorType.IsRestrictedType() && (IsDirectlyInIterator || IsInAsyncMethod()))
             {
-                diagnostics.Add(ErrorCode.ERR_BadSpecialByRefIterator, foreachKeyword.GetLocation(), getEnumeratorType);
+                CheckFeatureAvailability(foreachKeyword, MessageID.IDS_FeatureRefUnsafeInIteratorAsync, diagnostics);
```

Comment:
>CheckFeatureAvailability(foreachKeyword, MessageID.IDS_FeatureRefUnsafeInIteratorAsync, diagnostics); [](http://example.com/codeflow?start=16&length=101)

Are we testing scenario of a ref struct enumerator in ```await foreach```? That should still be an error regardless of language version and regardless of what we have in the body of the loop.

Reply:
Looks like we didn't have a test for ERR_BadSpecialByRefIterator, thanks.

Reply:
Yes, we have tests `RefIteratorInAsync` and `RefIteratorInIterator`. Agree the binding error should remain, will fix, thanks.